smoke-test.sh

```diff
@@ -6,8 +6,35 @@ VERSION_PREFIX=3.1
 # See https://github.com/dotnet/source-build/issues/579, this version
 # needs to be compatible with the runtime produced from source-build
 DEV_CERTS_VERSION_DEFAULT=3.0.0-preview8-28405-07
+
+# Use uname to determine what the CPU is.
+cpuName=$(uname -p)
+# Some Linux platforms report unknown for platform, but the arch for machine.
+if [[ "$cpuName" == "unknown" ]]; then
+  cpuName=$(uname -m)
+fi
+
+case $cpuName in
+  aarch64)
+    buildArch=arm64
+    ;;
+  amd64|x86_64)
+    buildArch=x64
+    ;;
+  armv*l)
+    buildArch=arm
+    ;;
+  i686)
+    buildArch=x86
+    ;;
+  *)
+    echo "Unknown CPU $cpuName detected, treating it as x64"
+    buildArch=x64
+    ;;
+esac
+
 __ROOT_REPO=$(cat "$SCRIPT_ROOT/artifacts/obj/rootrepo.txt" | sed 's/\r$//') # remove CR if mounted repo on Windows drive
-targetRid=$(cat "$SCRIPT_ROOT/artifacts/obj/x64/Release/TargetInfo.props" | grep -i targetrid | sed -E 's|\s*</?TargetRid>\s*||g')
+targetRid=$(cat "$SCRIPT_ROOT/artifacts/obj/${buildArch}/Release/TargetInfo.props" | grep -i targetrid | sed -E 's|\s*</?TargetRid>\s*||g')
```

Comment:
Could we pass this in from the MSBuild implementation from `./build.sh --run-smoke-tests` rather than re-reading and recalculating the arch here?

Reply:
I'm not sure it's worth going after this, I filed https://github.com/dotnet/source-build/issues/1654 about a broader revamp I think we should do. (Use a testing framework.)